internal/store/kubernetesdiscoverys/reducers.go

```diff
@@ -88,21 +89,28 @@ func RefreshKubernetesResource(state *store.EngineState, name string) {
 			}
 
 			krs.FilteredPods = r.FilteredPods
+			krs.Conditions = r.ApplyStatus.Conditions
 
 			isReadyOrSucceeded := false
-			for _, pod := range r.FilteredPods {
-				if krs.PodReadinessMode == model.PodReadinessSucceeded {
-					// for jobs, we don't care about whether it's ready, only whether it's succeeded
-					isReadyOrSucceeded = pod.Phase == string(v1.PodSucceeded)
-				} else {
-					isReadyOrSucceeded = len(pod.Containers) != 0 && store.AllPodContainersReady(pod)
-				}
-				if isReadyOrSucceeded {
-					// NOTE(nick): It doesn't seem right to update this timestamp everytime
-					// we get a new event, but it's what the old code did.
-					krs.LastReadyOrSucceededTime = time.Now()
+			if len(r.FilteredPods) != 0 {
+				for _, pod := range r.FilteredPods {
+					if krs.PodReadinessMode == model.PodReadinessSucceeded {
+						// for jobs, we don't care about whether it's ready, only whether it's succeeded
+						isReadyOrSucceeded = pod.Phase == string(v1.PodSucceeded)
+					} else {
+						isReadyOrSucceeded = len(pod.Containers) != 0 && store.AllPodContainersReady(pod)
+					}
 				}
+			} else {
+				isReadyOrSucceeded = meta.IsStatusConditionTrue(r.ApplyStatus.Conditions, v1alpha1.ApplyConditionJobComplete)
```

Comment:
it should be ok to do this check even if FilteredPods != 0

Reply:
Updated, there's technically still a length check but it's _after_ the status condition check (needed to handle the other change that makes us look at _all_ valid Pods instead of just the final one during iteration)